test/Microsoft.AspNet.Mvc.FunctionalTests/ModelBindingTests.cs

```diff
@@ -1327,5 +1327,42 @@ public class ModelBindingTests
             var body = await response.Content.ReadAsStringAsync();
             Assert.Equal(expectedContent, body);
         }
+
+        [Fact]
+        public async Task FormFileModelBinder_CanBind_SingleFile()
+        {
+            // Arrange
+            var server = TestServer.Create(_services, _app);
+            var client = server.CreateClient();
+            var url = "http://localhost/FileUpload/UploadSingle";
+            var streamContent = new StreamContent(new MemoryStream());
+            var formData = new MultipartFormDataContent("Upload----");
+            formData.Add(streamContent, "file", "test.txt");
+
+            // Act
+            var response = await client.PostAsync(url, formData);
+
+            // Assert
+            Assert.Equal(HttpStatusCode.Created, response.StatusCode);
+        }
+
+        [Fact]
+        public async Task FormFileModelBinder_CanBind_MultipleFiles()
+        {
+            // Arrange
+            var server = TestServer.Create(_services, _app);
+            var client = server.CreateClient();
+            var url = "http://localhost/FileUpload/UploadMultiple";
+            var streamContent = new StreamContent(new MemoryStream());
+            var formData = new MultipartFormDataContent("Upload----");
+            formData.Add(streamContent, "files", "test1.txt");
+            formData.Add(streamContent, "files", "test2.txt");
+
+            // Act
+            var response = await client.PostAsync(url, formData);
+
+            // Assert
+            Assert.Equal(HttpStatusCode.Created, response.StatusCode);
```

Comment:
This doesn't do enough in terms of testing the actual binding. Perhaps return a json payload which serializes the file content (or just their names)? Also would like to see at least one binding example where the object is nested inside a model so we know that scenario works too.